cloud/cluster/node_update.go

```diff
@@ -0,0 +1,39 @@
+package cluster
+
+import (
+	"fmt"
+)
+
+// NodeUpdate represents a change to the cluster
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+// Helper functions to create NodeUpdates
+
+func NewAdd(node Node) NodeUpdate {
+	return NodeUpdate{
+		NodeAdded,
+		node.Id(),
+		node,
+	}
+}
+
+func NewRemove(id NodeId) NodeUpdate {
+	return NodeUpdate{
+		UpdateType: NodeRemoved,
+		Id:         id,
+	}
+}
+
+type NodeUpdateSorter []NodeUpdate
```

Comment:
If a state receives a new state []Node with multiple changes, i.e. one node is missing and there's a new node included, we can create a diff against the old state, represented by a two element []NodeUpdate, but there's no guarantee it's correctly ordered. If we can guarantee we're only ever adding or removing one node at a time this isn't a problem but I don't know if we can make that guarantee.